cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -897,22 +917,33 @@ private boolean scanApplicationsWithReorder(ScanApplicationsRequest request,
   @Override
   public Map<ApplicationId, ApplicationMeta> getApplications(Collection<ApplicationId> ids) {
     return TransactionRunners.run(transactionRunner, context -> {
-      return getAppMetadataStore(context).getApplicationsForAppIds(ids);
+      Function<ApplicationId, SourceControlMeta> sourceControlRetriever = appId -> {
+        try {
+          return getSourceControlMetadataStore(context).get(appId);
+        } catch (IOException e) {
```

Comment:
We shouldn't ignore the exception in this case, as it would mistakenly be interpreted as the application not being in source control. It looks like we have a FunctionWithException class in cdap-common that would be a better fit here, letting it just throw the IOException

Reply:
Right. Thanks for pointing that out. I have made the change